docs/source/learn/eager.md

```diff
@@ -100,33 +118,45 @@ def step_fn(model, data, target, loss_fn, optimizer):
+because it lowers tracing overhead, leading to more efficient static inference
+graphs. To use it with XLA, simply specify `backend="openxla"`.
+
+``` python
+torch_xla.experimental.eager_mode(True)
+compiled_model = torch.compile(model, backend="openxla")
+```
+
+The long-term aim is for `torch.compile` to be the single compilation API for
+both training and inference on XLA.
 
-## Benchmark
+## Performance Benchmarks
 
-I run a 2 layer decoder only model training(it is pretty much just a
-llama2) with fake data on a single chip of v4-8 for 300 steps. Below is
-the number I observed.
+To quantify the performance impact of torch_xla.compile and eager mode,
+benchmarks were conducted under specific conditions. The benchmarks utilized a
+2-layer decoder-only model, similar to Llama2, trained with fake data. The
+training process spanned 300 steps on a single chip of a v4-8 TPU. The observed
+performance, measured in tokens per second, clearly illustrates the impact of
+different execution modes:
 
-  Mode                        token/s
-  --------------------------- ---------
-  Tracing mode (base line)    147
-  Eager mode                  65
-  Eager + torch_xla compile   147
+| Mode                        | token/s |
+|-----------------------------|---------|
+| Tracing mode (base line)    | 147     |
+| Eager mode                  | 65      |
+| Eager + torch_xla compile   | 147     |
 
-  : Eager mode benchmarks
+Eager mode with `torch_xla.compile` matches the performance of traditional
+LazyTensor tracing mode at `147` tokens/s, demonstrating a better user
+experience without performance loss.
 
-Eager mode can achieve ~45% performance of the fully compiled model for
-the decoder only model. For more information, see
-[train_decoder_only_base.py](https://github.com/pytorch/xla/blob/master/examples/train_decoder_only_base.py)
+Pure eager mode's performance is model-dependent; it achieves ~45% of the fully
+compiled model's performance for decoder-only models. However, for ResNet50,
+pure eager mode was significantly slower (about 1% of compiled mode). For more
+information, see [train_decoder_only_base.py](https://github.com/pytorch/xla/blob/master/examples/train_decoder_only_base.py)
 and [eager example](https://github.com/pytorch/xla/tree/master/examples/eager).
-Note that perfomrane of the eager mode is very model dependent. When I
-tried to run the resnet50, the eager mode perfomrance is \~1% of the
-compiled mode. We don't exepct user to use eager mode to execute the
-main training loop. Eager mode is meant to be used to handle non-core
-part of the training/inference logic(Data preprocessing, random number
-generations etc) or debug.
+This varying overhead means pure eager mode is not intended for main training or
+inference loops. Its utility lies in non-core tasks like data preprocessing,
+random number generation, custom utilities, or debugging, where immediate
+execution is prioritized over throughput.
```

Comment:
We also need to call out that `torch_xla.compile` is independently useful, even not in eager mode. That's why torchprime does: https://github.com/AI-Hypercomputer/torchprime/blob/31c450e82c6273f50f9815351f6fbebb42903f58/torchprime/torch_xla_models/train.py#L330

Wrapping the training loop in `torch_xla.compile` provides a few benefits:
- There's no `mark_step` anywhere
- Dataloading operations don't leak into the training loop graph. This benefit is similar to if eager mode was turned on. The only difference is that the dataloading operations are captured into a separate graph as opposed to running eagerly.
- `torch_xla.compile(full_graph=True)` will catch accidental graph breaks

Reply:
LGTM